packages/ra-core/src/form/useApplyInputDefaultValues.ts

```diff
@@ -58,11 +62,11 @@ export const useApplyInputDefaultValues = ({
         // Since we use get(record, source), if source is like foo.23.bar,
         // this effect will run. However we only want to set the default value
         // for the subfield bar if the record actually has a value for foo.23
-        const pathContainsIndex = source
+        const pathContainsIndex = finalSource
             .split('.')
             .some(pathPart => numericRegex.test(pathPart));
         if (pathContainsIndex) {
-            const parentPath = source.split('.').slice(0, -1).join('.');
+            const parentPath = finalSource.split('.').slice(0, -1).join('.');
```

Comment:
can't we get the parent path simply by calling the SourceContext?

Reply:
The SourceContext does not give the parent path, it returns the modified source for its children inputs/fields. Did I miss something?